### ISSUE TITLE
Create async image optimization process

### ISSUE THREAD

**Issue by mattstratton** [devopsdays/devopsdays-web#8133]
Since we have removed image processing/cropping from the build, we should have some kind of async script that resizes images and optimizes them.

Ideally this would be a script that would run in a serverless thing, or on an ec2 instance on a schedule, and it would then push in a PR with its optimizations.

**Comment by toshywoshy** [comment]
Why not use the Hugo builtin image processor and optimizer?

**Comment by mattstratton** [comment]
I’ve actually started looking at this a little bit, it’s harder than you think because you have to change the whole way that the images are reasoned about, however, I think it works by having a short code that people can use in their markdown files that will optimize the images, and then we can also use that optimization in the templates that have images with them always… I think I started writing this issue before Hugo pipes were a thing!

**Comment by toshywoshy** [comment]
I am sure your usage predates the HugoIO image optimization
As I also had to rewrite the image processing for CfgMgmtCamp and DoD usage of HugoIO predates that

If you want to see an example, here is a simple example of taking an image and rendering it to WebP output.
https://github.com/cfgmgmtcamp/cfgmgmtcamp.eu/blob/master/themes/cfgmgmtcamp/layouts/sponsors/list.html
Lines 29:34

**Comment by mattstratton** [comment]
Yep, I definitely know how to do it as I’ve done it and other Hugo themes, a lot of this goes into some other improvements I want to make that would require some additional parts of the tool chain besides, just the Hugo executable, but I think that we can figure out a way to make that work!
